apps/k8s-io/configmap-nginx.yaml

```diff
@@ -39,6 +39,22 @@ data:
           }
 
           if ($arg_go-get = "1") {
+            # special case for experimenting with "k8s.io/internal/*" to "github.com/k8s-internal/*"
+            if ($repo = "internal" {
+              # This is a go-get operation.
+              return 200 '
+                  <html><head>
+                        <meta name="go-import"
+                              content="k8s.io/$repo
+                                       git https://github.com/k8s-internal/$repo">
```

Comment:
This will send you to https://github.com/k8s-internal/internal, which isn't right.

Given `go get k8s.io/internal/foo` we want to produce `github.com/k8s-internal/foo` right?  But the regex in `location` only parses the first token ("internal" here) and "the rest" (optional).  So you might think to do `https://github.com/k8s-internal/$subpath` but that doesn't work because `go get k8s.io/internal/foo/bar` is a valid request but the result must still be `github.com/k8s-internal/foo` (no path element).

So I think we have 2 choices.

1) Make the regex more complicated, like `^/(?<path1>[^/]*)(?<path2>[^/])?(?<subpath>/.*)?$` (I am not sure that is right, we'd have to canary it and see :) and then write the two content-blocks to interpret `$path1` as `repo` in the normal case, but if `$path1` is "internal" then `$path2` is `repo`.  

2) Copy the whole `location` block, and change the regex like `^/internal/(?<repo>[^/]*)(?<subpath>/.*)?$` and then `$repo` is correct in both blocks.

#2 seems simpler to me.  It's a little duplicative, but I can comprehend that more easily than a regex.

Note that you need to put the more specific "internal" match first, of course :)

I would try #2, make the only difference be the regex and the github URLs, push to canary, then:

`curl -i -k -H hostname=k8s.io 'https://34.102.239.89/foo/bar?go-get=1`

...and:

`curl -i -k -H hostname=k8s.io 'https://34.102.239.89/internal/foo/bar?go-get=1`

...and see what it produces.

To wrap this up, we should make a repo at github/kubernetes/internal and put a README that it's just a place-holder and that `go get` should send you to k8s-internal.

Reply:
updated PR to do #2